libs/aml/src/amlParser.ts

```
@@ -722,25 +652,29 @@ function tokenInfo(token: IToken, issues?: TokenIssue[]): TokenInfo {
+        offset: {start: pos(token.startOffset), end: pos(token.endOffset)},
         position: {
-            start: {line: token.startLine || defaultPos, column: token.startColumn || defaultPos},
-            end: {line: token.endLine || defaultPos, column: token.endColumn || defaultPos}
+            start: {line: pos(token.startLine), column: pos(token.startColumn)},
+            end: {line: pos(token.endLine), column: pos(token.endColumn)}
         }
     }
 }
 
+function pos(value: number | undefined): number {
+    return value !== undefined && !isNaN(value) ? value : defaultPos
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)